### PR TITLE
Use `delegate` to call the methods to `@conn`

### PR DESCRIPTION
**Pull request by kamipo**

### CODE REVIEW

**Comment by arthurnn** [comment]
I am not sure what are the gains of this PR.
Is there any performance improvement I am missing?

thanks anyways @kamipo

File: activerecord/lib/active_record/connection_adapters/abstract/schema_creation.rb [hunk 1]
```diff
@@ -18,6 +18,9 @@ def visit_AddColumn(o)
           "ADD #{accept(o)}"
         end
 
+        delegate :quote_column_name, :quote_table_name, :quote_default_expression, :type_to_sql, to: :@conn
```
Comment (egilburg) [inline activerecord/lib/active_record/connection_adapters/abstract/schema_creation.rb]: If you just move this below the `private` statement, do you still need separate line to mark these as private?

**Comment by egilburg** [comment]
@arthurnn seems like more dry code, also reduces burden of having to maintain method signature compatibility between the methods on this class and those on `@conn`

File: activerecord/lib/active_record/connection_adapters/abstract/schema_creation.rb (hunk 1 above)
Comment (kamipo) [inline activerecord/lib/active_record/connection_adapters/abstract/schema_creation.rb]: Yes, still need, a method that has been defined by `delegate` will be public, regardless of the visibility of the current scope.

**Comment by kamipo** [comment]
@arthurnn Thanks @egilburg for your explain. This PR is for more dry code.

**Comment by vipulnsward** [comment]
LGTM

**Comment by tenderlove** [comment]
Quoting column / table names is a hot spot.  Using the meta-programmed delegate method adds overhead that we do not need.

**Comment by kamipo** [comment]
The quoting methods in `SchemaCreation` are used only in the migration. I think the migration is not a hot spot of the performance, so I think it is worth making it more dry code. Do you think the migration is a hot spot?

**Comment by sgrif** [comment]
Regardless, this qualifies as a cosmetic change, which we generally do not accept.

**Pull request closed without merging**